subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/UnexportMainSymbol.java

```diff
@@ -82,26 +83,37 @@ public DirectoryProperty getOutputDirectory() {
     public void unexport() {
         for (final File file: source) {
             final File relocatedObject = outputDirectory.file(file.getName()).get().getAsFile();
-            getProject().exec(new Action<ExecSpec>() {
-                @Override
-                public void execute(ExecSpec execSpec) {
-                    // TODO: should use target platform to make this decision
-                    if (OperatingSystem.current().isMacOsX()) {
-                        execSpec.executable("ld"); // TODO: Locate this tool from a tool provider
-                        execSpec.args(file);
-                        execSpec.args("-o", relocatedObject);
-                        execSpec.args("-r"); // relink, produce another object file
-                        execSpec.args("-unexported_symbol", "_main"); // hide _main symbol
-                    } else if (OperatingSystem.current().isLinux()) {
-                        execSpec.executable("objcopy"); // TODO: Locate this tool from a tool provider
-                        execSpec.args("-L", "main"); // hide main symbol
-                        execSpec.args(file);
-                        execSpec.args(relocatedObject);
-                    } else {
-                        throw new IllegalStateException("Do not know how to unexport a main symbol on " + OperatingSystem.current());
+            if (OperatingSystem.current().isWindows()) {       
+                try {
+                    final SymbolHider symbolHider = new SymbolHider(file);
+                    symbolHider.hideSymbol("main");     // 64 bit
+                    symbolHider.hideSymbol("_main");    // 32 bit
+                    symbolHider.saveTo(relocatedObject);
+                } catch (IOException e) {
+                    throw new IllegalStateException("Failed to unexport a main symbol on " + OperatingSystem.current());
```

Comment:
So could do with some advise here on what the best approach to handling the possibility of an IOException exception is, the current approach of matching the same sort of exception as on linux and mac seems a bit hacky.

Reply:
Just use `throw UncheckedException.throwAsUncheckedExeception(e)` here.  That's an internal method that idiomatically wraps the exception in an unchecked exception that bubbles up in a way that Gradle can consume.

Reply:
Done, thanks